### ISSUE TITLE
Incorrect behavior of Array#slice!(start, length)

### ISSUE THREAD

**Issue by ruipserra** [rubinius/rubinius#3321]
I ran into this issue with the Array#slice! method:

```
a = [1]         # => [1]
a.slice!(0, 1)  # => [1]
a[-1]           # => 1
a[0]            # => nil
a               # => []
```

This happens when the `length` argument is 0 or any positive value. On the other hand, if `length` is negative, we get the following:

```
a = [1]         # => [1]
a.slice!(0, -1) # => Rubinius::ObjectBoundsExceededError: Tuple::copy_from: index -1 out of bounds for size 1
```
whereas MRI returns nil.

Rubinius version:
`rubinius 2.5.2 (2.1.0 7a5b05b1 2015-01-30 3.5.0 JI) [x86_64-darwin13.4.0]`

OS:
`Darwin Kernel Version 13.4.0: Sun Aug 17 19:50:11 PDT 2014; root:xnu-2422.115.4~1/RELEASE_X86_64 x86_64 i386 MacBookPro9,2 Darwin`

**Comment by brixen** [comment]
@ruipserra just FYI, if you add eg `Fixes #3321` to your PR commit message, when the PR is merged, the issue automatically gets closed.

**Comment by ruipserra** [comment]
@brixen thanks, I didn't know that.
